source/Generic/GamesSizeCalculator/Steam/SteamSizeCalculator.cs

```diff
@@ -9,84 +10,160 @@
-            var biggestDepotsCopy = new List<DepotInfo>(biggestDepots);
-            foreach (var depot in biggestDepots)
+            var parsedDepots = allDepots.Select(ParseDepotName).ToList();
+            var blacklistedDepots = parsedDepots.Where(d => RegionalWordsBlacklist.Contains(d.RegionWord, StringComparer.InvariantCultureIgnoreCase)).ToList();
+            foreach (var d in blacklistedDepots)
             {
-                var lastWordInName = GetLastWord(depot.Name)?.ToLowerInvariant();
-                switch (lastWordInName)
+                logger.Trace($"Removing depot {d.BaseName} due to blacklisted region word: {d.RegionWord}");
+                parsedDepots.Remove(d);
+                allDepots.Remove(d.Depot);
+            }
+            var grouped = parsedDepots.GroupBy(d => d.BaseName);
+
+            foreach (var group in grouped)
+            {
+                var key = group.Key;
+                var orderedDepots = group.OrderBy(d => d.Depot.Optional).ThenBy(d => d.RegionWord == null ? -2 : Array.IndexOf(RegionalWords, d.RegionWord)).ToList();
+                if (orderedDepots.Count == 1)
+                {
+                    continue;
+                }
+
+                string logString = $"Depot group {key}, {orderedDepots.Count} depots:";
+
+                for (int i = 0; i < orderedDepots.Count; i++)
                 {
-                    case null:
-                        break;
-                    case "eu":
-                    case "europe":
-                    case "asia":
-                    case "aus":
-                    case "australia":
-                    case "nz":
-                    case "usa":
-                    case "us":
-                    case "ru":
-                    case "russia":
-                    case "germany":
-                    case "deutschland":
-                    case "de":
-                    case "es":
-                    case "sa":
-                    case "cn":
-                    case "china":
-                    case "row":
-                    case "ww":
-                        if (biggestDepotsCopy.Count == 1) // Don't remove the last big depot
-                        {
-                            return;
-                        }
-
-                        biggestDepotsCopy.Remove(depot);
-                        depots.Remove(depot);
-                        logger.Debug($"Removed depot {depot.Name}");
-                        break;
-                    default:
-                        break;
+                    var depotData = orderedDepots[i];
+                    string depotLogString = $"{depotData.Depot.Id} - {depotData.Depot.Name}, optional: {depotData.Depot.Optional}, dlc: {depotData.Depot.IsDLC}, size: {depotData.Depot.FileSize}, {depotData.BaseName}|{depotData.RegionWord}|{depotData.Rank}";
+
+                    if (i == 0)
+                    {
+                        logString += "Keeping depot: " + depotLogString;
```

Comment:
Imo a StringBuilder should be used instead of recreating the string with each addition